File: app/src/main/java/com/checkin/app/checkin/Account/AccountViewModel.java
package com.checkin.app.checkin.Account;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.checkin.app.checkin.Data.BaseViewModel;
import com.checkin.app.checkin.Data.Resource;
import com.checkin.app.checkin.Utility.SourceMappedLiveData;

import java.util.List;

public class AccountViewModel extends BaseViewModel {
    private AccountRepository mRepository;
    private SourceMappedLiveData<Resource<List<AccountModel>>> mAccounts = createNetworkLiveData();
    private MutableLiveData<AccountModel> mCurrentAccount = new MutableLiveData<>();

    public AccountViewModel(@NonNull Application application) {
        super(application);
        mRepository = AccountRepository.getInstance(application);
    }

    public LiveData<Resource<List<AccountModel>>> getAccounts() {
        fetchAccounts();
        return mAccounts;
    }

    private void fetchAccounts() {
        mAccounts.addSource(mRepository.getSelfAccounts(), mAccounts::setValue);
    }

    public LiveData<AccountModel> getCurrentAccount() {
        return mCurrentAccount;
    }

    public void setCurrentAccount(AccountModel account) {
        mCurrentAccount.setValue(account);
    }

    @Override
    public void updateResults() {
        fetchAccounts();
    }
}

File: app/src/main/java/com/checkin/app/checkin/Account/AccountViewModel.kt
package com.checkin.app.checkin.Account

import android.app.Application
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.checkin.app.checkin.data.BaseViewModel
import com.checkin.app.checkin.data.resource.Resource

class AccountViewModel(application: Application) : BaseViewModel(application) {
    private val mRepository: AccountRepository = AccountRepository.getInstance(application)

    private val mAccounts = createNetworkLiveData<List<AccountModel>>()
    private val mCurrentAccount = MutableLiveData<AccountModel>()
    val accounts: LiveData<Resource<List<AccountModel>>>
        get() {
            fetchAccounts()
            return mAccounts
        }

    fun fetchAccounts() {
        if (mAccounts.value?.status != Resource.Status.SUCCESS)
            mAccounts.addSource(mRepository.selfAccounts, mAccounts::setValue)
    }

    val currentAccount: LiveData<AccountModel> = mCurrentAccount

    fun setCurrentAccount(account: AccountModel) {
        mCurrentAccount.value = account
    }

    override fun updateResults() {
        fetchAccounts()
    }
}